pkg/controllers/networkpolicy/sync.go

```diff
@@ -281,24 +287,19 @@ func (r *NetworkPolicyReconciler) getNICsOfCloudResources(resources []*securityg
 
 // checkAndUpdateIndexer checks if rule is present in indexer and updates the indexer if not present.
 // Returns true if indexer is updated.
-func (a *appliedToSecurityGroup) checkAndUpdateIndexer(r *NetworkPolicyReconciler, rule securitygroup.Rule,
+func (a *appliedToSecurityGroup) checkAndUpdateIndexer(r *NetworkPolicyReconciler, rule *securitygroup.CloudRule,
 	existingRuleMap map[string]*securitygroup.CloudRule) bool {
 	indexerUpdate := false
 
-	// deep copy the rule and construct CloudRule object from it.
-	ruleCopy := deepcopy.Copy(rule).(securitygroup.Rule)
-	cr := &securitygroup.CloudRule{
-		Rule:         ruleCopy,
-		AppliedToGrp: a.id.CloudResourceID.String(),
-	}
-	cr.Hash = cr.GetHash()
+	// deep copy the rule.
+	ruleCopy := deepcopy.Copy(rule).(*securitygroup.CloudRule)
```

Comment:
why rule copy is required here?

Reply:
It can be removed by passing object into func instead of pointers, done.